pinot-core/src/main/java/org/apache/pinot/core/segment/index/converter/SegmentV1V2ToV3FormatConverter.java

```diff
@@ -200,8 +200,9 @@ private void readCopyBuffers(SegmentDirectory.Reader reader, SegmentDirectory.Wr
       ColumnIndexType indexType)
       throws IOException {
     PinotDataBuffer oldBuffer = reader.getIndexFor(column, indexType);
-    PinotDataBuffer newDictBuffer = writer.newIndexFor(column, indexType, oldBuffer.size());
-    oldBuffer.copyTo(0, newDictBuffer, 0, oldBuffer.size());
+    PinotDataBuffer newBuffer = writer.newIndexFor(column, indexType, oldBuffer.size());
+    oldBuffer.copyTo(0, newBuffer, 0, oldBuffer.size());
+    newBuffer.flush();
```

Comment:
We should flush the buffer in SegmentLocalFSDirectory.save() which is empty right now.

Reply:
correct, removed this line.